### PR TITLE
Add Typescript definitions for 'create-emotion-server' and 'emotion-server'

### PR DESCRIPTION
**Pull request by mgroenhoff**
**What**: Typescript definitions added for `create-emotion-server` and `emotion-server`

**Why**: Because Typescript!

**How**:

**Checklist**:
- [ ] Documentation N/A
- [x] Tests
- [ ] Code complete N/A

### CODE REVIEW

**Comment by mgroenhoff** [comment]
Please update `yarn.lock` and commit it yourself because I can't seem to get it to generate the correct registry (https://github.com/yarnpkg/yarn/issues/3330)

File: packages/emotion-server/types/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dtslint.json",
+    "rules": {
+        "no-relative-import-in-test": false
+    }
+}
```
Comment (Ailrun) [inline packages/emotion-server/types/tslint.json]: You miss the final linefeeding for this file

File: packages/emotion-server/package.json [hunk 2]
```diff
@@ -20,8 +23,10 @@
     "emotion": "^9.1.3"
   },
   "devDependencies": {
+    "@types/react-dom": "16.0.5",
```
Comment (Ailrun) [inline packages/emotion-server/package.json]: It should depend on `@types/react-dom` as dependencies, or, at least as peerDependencies, shouldn't it?

Comment (mgroenhoff) [reply packages/emotion-server/package.json]: The only use is inside the tests so I don't think it should.

Comment (Ailrun) [reply packages/emotion-server/package.json]: @mgroenhoff Oh, you're right. My miss.

**Comment by mgroenhoff** [comment]
Thanks!

**Pull request merged**